src/Illuminate/Database/Query/Builder.php

```diff
@@ -1210,6 +1227,10 @@ public function whereYear($column, $operator, $value = null, $boolean = 'and')
             $value, $operator, func_num_args() === 2
         );
 
+        if ($value instanceof Carbon) {
+            $value = $value->format('Y');
```

Comment:
We can use `$value->year`.

Reply:
Just learnt about this, that's neat. However it won't work the same for day and month because they aren't zero padded. I think it might be better to be consistent with the use of the `format` method rather than doing it differently in one place.

Reply:
If we go back to just using `DateTime` I can just replace the uses of `toDateString` and `toTimeString` with calls to `format` as well and then they'll all be consistent. Curious to hear what people think about that.

Reply:
You could say the same about `toDateString()` and `toTimeString()` ;-)